src/test/java/org/opensearch/knn/index/codec/KNN940Codec/KNN940CodecTests.java

```diff
@@ -26,15 +22,13 @@ public void testBuildFromModelTemplate() throws InterruptedException, ExecutionE
         testBuildFromModelTemplate((KNN940Codec.builder().delegate(V_9_4_0.getDefaultCodecDelegate()).build()));
     }
 
-    public void testKnnVectorIndex() throws Exception {
-        Function<MapperService, PerFieldKnnVectorsFormat> perFieldKnnVectorsFormatProvider = (
-            mapperService) -> new KNN940PerFieldKnnVectorsFormat(Optional.of(mapperService));
-
-        Function<PerFieldKnnVectorsFormat, Codec> knnCodecProvider = (knnVectorFormat) -> KNN940Codec.builder()
+    // Ensure that the codec is able to return the correct per field knn vectors format for codec
+    public void testCodecSetsCustomPerFieldKnnVectorsFormat() {
+        final Codec codec = KNN940Codec.builder()
             .delegate(V_9_4_0.getDefaultCodecDelegate())
-            .knnVectorsFormat(knnVectorFormat)
+            .knnVectorsFormat(V_9_4_0.getPerFieldKnnVectorsFormat())
             .build();
 
-        testKnnVectorIndex(knnCodecProvider, perFieldKnnVectorsFormatProvider);
+        assertTrue(codec.knnVectorsFormat() instanceof KNN940PerFieldKnnVectorsFormat);
```

Comment:
Maybe use class of format class the is defined in the version enum, something like:
```
assertThat(codec.knnVectorsFormat(), isA(V_9_4_0.getPerFieldKnnVectorsFormat().getClass()));
```

Reply:
I beliebe that assertThat is deprecated

Reply:
org.hamcrest.MatcherAssert.assertThat should be ok